src/main/java/com/microsoft/azure/functions/worker/binding/RpcHttpRequestDataSource.java

```diff
@@ -22,8 +25,10 @@ public RpcHttpRequestDataSource(String name, RpcHttp value) {
 		super(name, null, HTTP_DATA_OPERATIONS);
 		this.httpPayload = value;
 		this.bodyDataSource = BindingDataStore.rpcSourceFromTypedData(null, this.httpPayload.getBody());
-		this.fields = Arrays.asList(this.httpPayload.getHeadersMap(), this.httpPayload.getQueryMap(),
-				this.httpPayload.getParamsMap());
+		this.fields = Arrays.asList(
+				convertFromNullableMap(this.httpPayload.getNullableHeadersMap()),
+				convertFromNullableMap(this.httpPayload.getNullableQueryMap()),
+				convertFromNullableMap(this.httpPayload.getNullableParamsMap()));
```

Comment:
not related to this PR, but do we know difference between query map and param map?

Reply:
Looking at the host code, it looks like the three key differences between [query map](https://github.com/Azure/azure-functions-host/blob/11a0add2eb0c14588e5070aed3d0489782762be8/src/WebJobs.Script.Grpc/MessageExtensions/GrpcMessageConversionExtensions.cs#L116-L130) and [param map](https://github.com/Azure/azure-functions-host/blob/11a0add2eb0c14588e5070aed3d0489782762be8/src/WebJobs.Script.Grpc/MessageExtensions/GrpcMessageConversionExtensions.cs#L149-L166) are that:
1. They come from different parts of the `request` object (i.e. `request.Query` vs `request.HttpContext.Items.TryGetValue(HttpExtensionConstants.AzureWebJobsHttpRouteDataKey, out object routeData)`)
2. We do not add null values to the parameters map, even if the `UseNullableValueDictionaryForHttp` capability is on 
3. We add empty strings to the parameters map, even if the `UseNullableValueDictionaryForHttp` capability is off